src/kakarot/instructions/environmental_information.cairo

```diff
@@ -487,42 +483,69 @@ namespace EnvironmentalInformation {
             contract_address=registry_address_, evm_contract_address=address_felt
         );
 
-        // handle case where there is no eth -> stark address mapping
-        if (starknet_contract_address == 0) {
-            return ctx;
-        }
 
-        // Get the bytecode from the Starknet_contract
-        let (bytecode_len, bytecode) = IEvmContract.bytecode(
-            contract_address=starknet_contract_address
-        );
-
-        // handle case were eth address returns no bytecode:
+        if (starknet_contract_address != 0) {
+            //  we get the bytecode from the Starknet_contract
+            let (bytecode_len, bytecode) = IEvmContract.bytecode(
+                contract_address=starknet_contract_address
+            ); 
+
+            tempvar syscall_ptr=syscall_ptr;
+            tempvar pedersen_ptr=pedersen_ptr;
+            tempvar range_check_ptr=range_check_ptr;                        
+            tempvar bitwise_ptr=bitwise_ptr;
+            tempvar bytecode_len=bytecode_len;
+        } else {
+            // otherwise handle case where there is no eth -> stark address mapping            
+            let bytecode : felt* = alloc();
+            let bytecode_len = 0;
+
+            tempvar syscall_ptr=syscall_ptr;
+            tempvar pedersen_ptr=pedersen_ptr;
+            tempvar range_check_ptr=range_check_ptr;                        
+            tempvar bitwise_ptr=bitwise_ptr;
+            tempvar bytecode_len=bytecode_len;            
 
-        if (bytecode_len == 0) {
-            return ctx;
         }
 
-        // TODO do we have the distinction between precompiles and warm and cold addresses?
+        tempvar syscall_ptr=syscall_ptr;
+        tempvar pedersen_ptr=pedersen_ptr;
+        tempvar range_check_ptr=range_check_ptr;                        
+        tempvar bitwise_ptr=bitwise_ptr;
+        tempvar bytecode_len=bytecode_len;            
 
+    
         // Get bytecode slice from offset to size
+        // in the case were 
+        // eth address returns no bytecode or has no `starknet_contract_address`
+        // the bytecode len would be zero and the byte array empty, 
+        // which would `Helpers.slice_data` would return an array
+        // with the requested `size` of zeroes
+
         let sliced_bytecode: felt* = Helpers.slice_data(
             data_len=bytecode_len, data=bytecode, data_offset=offset.low, slice_len=size.low
         );
 
         // Write bytecode slice to memory at dest_offset
-        // TODO handle dynamic gas for memory expansion
+        let (memory, memory_expansion_cost) = Memory.ensure_length(self=ctx.memory, length=dest_offset.low + size.low);
         let memory: model.Memory* = Memory.store_n(
```

Comment:
when one writes to memory, there is always (AFAIK) a possibility to trigger memory expansion cost. So it's relevant to `ensure_length` here

> During an execution, the whole memory is accessible, but not for free. When an offset is accessed for the first time (either read or write), it may trigger a memory expansion, which will cost gas